### ISSUE TITLE
Setting different n_cpus in DefaultInference gives results in the same time

### ISSUE THREAD

**Issue by yihming** [owkin/PyDESeq2#288]
**Describe the bug**
I tried setting different `n_cpus` value for `DefaultInference` (e.g. 1 and 8):
```python
from pydeseq2.default_inference import DefaultInference
inference = DefaultInference(n_cpus=n_cpus)
```
Then `dds.deseq2()` and `DeseqStats.summary()` give the similar time (~32s) to finish for the same data.

**To Reproduce**
Provide snippets of code and steps on how to reproduce the behavior. 
Please also specify the version you are using.

**Expected behavior**

I wonder if I didn't set the parameter correctly in order to speed up the computation by parallel computing. Or if I did set it correctly, why the calculation did not take effects. Thanks!

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Ubuntu 22.04.4 LTS with 8 vCPUs available
 - Version: pydeseq2 v0.4.9

**Additional context**

**Comment by abearab** [comment]
I can confirm I also experienced the same issue.

**Comment by umarteauowkin** [comment]
Hi ! Thanks for reporting this. Could you tell me the number of genes with which you experience this issue ?

**Comment by yihming** [comment]
Hi @umarteauowkin ,

Thanks for checking this issue. I tried your suggestion, and found that the Inference object's `n_cpus` is always overwritten to use all available vCPUs.

I tried to fix the issue in PR #293 . It works at my side. Please let me know if I need to do anything further. Thanks!

**Comment by umarteauowkin** [comment]
Solved by #293. Thanks @yihming !
